src/store/writer.rs

```diff
@@ -34,10 +35,16 @@ impl StoreWriter {
     ///
     /// The store writer will writes blocks on disc as
     /// document are added.
-    pub fn new(writer: WritePtr, compressor: Compressor, block_size: usize) -> StoreWriter {
+    pub fn new(
+        writer: WritePtr,
+        compressor: Compressor,
```

Comment:
why isn't that part of the compressor?

Reply:
Compressor is currently constructed by deserializing from the compressor name. To pass more parameters we would need much more complicated deserialization.
And there are cases where the Compressor is only used for decompression.